Dockerfile

```diff
@@ -27,14 +27,15 @@ ARG PATCH_LIST
 ARG INSTALL_TYPE=source
 
 # build arguments for source build customization
+ARG UPPER_CONSTRAINTS_FILE=upper-constraints.txt
 ARG IRONIC_SOURCE
 ARG IRONIC_INSPECTOR_SOURCE
 ARG IRONIC_LIB_SOURCE
 ARG SUSHY_SOURCE
 
 COPY sources /sources/
 
-COPY ironic-${INSTALL_TYPE}-list ${PKGS_LIST} ${EXTRA_PKGS_LIST:-$PKGS_LIST} ${PATCH_LIST:-$PKGS_LIST} /tmp/
+COPY ${UPPER_CONSTRAINTS_FILE} ironic-${INSTALL_TYPE}-list ${PKGS_LIST} ${EXTRA_PKGS_LIST:-$PKGS_LIST} ${PATCH_LIST:-$PKGS_LIST} /tmp/
```

Comment:
What I meant: if you default UPPER_CONSTRAINTS_FILE to an empty string, this line should just work as it was before the change.

Reply:
I'm really having a hard time following this.
How do I copy the file inside the image if I don't modify this line?
What I was thinking was actually removing the need of UPPER_CONSTRAINTS_FILE variable and just use the name upper-constraints.txt for simplicity

Reply:
You can modify this line, just the file will be empty?
```bash
$ FOO=
$ echo $FOO a b c
a b c
```

Reply:
if I remember correctly there was an issue with that, but I'll give it a try